.github/workflows/src/summarize-checks/labelling.js

```diff
@@ -23,6 +72,220 @@ import {
+ */
+
+/**
+ * This file is the single source of truth for types used by the OpenAPI specification breaking change checks
+ * in the Azure/azure-rest-api-specs and Azure/azure-rest-api-specs-pr repositories.
+ *
+ * For additional context, see:
+ *
+ * - "Deep-dive into breaking changes on spec PRs"
+ *   https://aka.ms/azsdk/pr-brch-deep
+ *
+ * - "[Breaking Change][PR Workflow] Use more granular labels for Breaking Changes approvals"
+ *   https://github.com/Azure/azure-sdk-tools/issues/6374
+ */
+/**
+ * @typedef {"SameVersion" | "CrossVersion"} BreakingChangesCheckType
+ * @typedef {"VersioningReviewRequired" | "BreakingChangeReviewRequired"} ReviewRequiredLabel
+ * @typedef {"Versioning-Approved-*" | "BreakingChange-Approved-*"} ReviewApprovalPrefixLabel
+ * @typedef {"Versioning-Approved-Benign" | "Versioning-Approved-BugFix" | "Versioning-Approved-PrivatePreview" | "Versioning-Approved-BranchPolicyException" | "Versioning-Approved-Previously" | "Versioning-Approved-Retired"} ValidVersioningApproval
+ * @typedef {"BreakingChange-Approved-Benign" | "BreakingChange-Approved-BugFix" | "BreakingChange-Approved-UserImpact" | "BreakingChange-Approved-BranchPolicyException" | "BreakingChange-Approved-Previously" | "BreakingChange-Approved-Security"} ValidBreakingChangeApproval
+ * @typedef {ReviewRequiredLabel | ReviewApprovalPrefixLabel | ValidBreakingChangeApproval | ValidVersioningApproval} SpecsBreakingChangesLabel
+ */
+/**
+ * @typedef {Object} BreakingChangesCheckConfig
+ * @property {ReviewRequiredLabel} reviewRequiredLabel
+ * @property {ReviewApprovalPrefixLabel} approvalPrefixLabel
+ * @property {(ValidVersioningApproval | ValidBreakingChangeApproval)[]} approvalLabels
+ * @property {string} [deprecatedReviewRequiredLabel]
+ */
+
+/**
+ * @typedef {"SwaggerFile" | "TypeSpecFile" | "ExampleFile" | "ReadmeFile"} FileTypes
+ */
+
+/**
+ * @typedef {"Addition" | "Deletion" | "Update"} ChangeTypes
+ */
+
+/**
+ * @typedef {Object} PRChange
+ * @property {FileTypes} fileType
+ * @property {ChangeTypes} changeType
+ * @property {string} filePath
+ * @property {any} [additionalInfo]
+ */
+
+/**
+ * @typedef {Object} ChangeHandler
+ * @property {function(PRChange): void} [SwaggerFile]
+ * @property {function(PRChange): void} [TypeSpecFile]
+ * @property {function(PRChange): void} [ExampleFile]
+ * @property {function(PRChange): void} [ReadmeFile]
+ */
+
+/**
+ * @typedef {"resource-manager" | "data-plane"} PRType
+ */
+
+/**
+ * Represents a GitHub label.
+ * Currently used in the context of processing
+ * labels related to the review workflow of PRs submitted to
+ * Azure/azure-rest-api-specs and Azure/azure-rest-api-specs-pr repositories.
+ * This processing happens in the prSummary.ts / summary() function.
+ *
+ * See also: https://aka.ms/SpecPRReviewARMInvariants
+ */
+// todo: inject `core` for access to logging
+export class Label {
```

Comment:
Our convention is to:

1. Prefix private fields with `#` (makes them private at runtime)
2. Declare fields (with types) above ctor

Example:

https://github.com/Azure/azure-rest-api-specs/blob/a12ced1212df32affe4dd203cf7d99dd71acc985/.github/shared/src/spec-model.js#L20-L29